spec/requests/api/graphql/queries/offer_spec.rb

```diff
@@ -155,5 +200,52 @@
         expect(error_message).to eq 'Offer not found'
       end
     end
+
+    context 'with a draft offer' do
+      before { offer.update!(state: Offer::DRAFT) }
+
+      context 'as an admin' do
+        let(:token) do
+          payload = { aud: 'gravity', sub: 'adminid', roles: 'admin' }
+          JWT.encode(payload, Convection.config.jwt_secret)
+        end
+
+        it 'returns that offer' do
+          post '/api/graphql', params: { query: query }, headers: headers
+
+          expect(response.status).to eq 200
+          body = JSON.parse(response.body)
+
+          offer_response = body.dig('data', 'offer')
+          expect(offer_response).to match(
+            {
+              'id' => offer.id.to_s,
+              'commissionPercentWhole' => offer.commission_percent_whole
+            }
+          )
+        end
+      end
+
+      context 'as a valid user' do
+        let(:token) do
+          user_id = offer.submission.user.gravity_user_id
+          payload = { aud: 'gravity', sub: user_id, roles: 'user' }
+          JWT.encode(payload, Convection.config.jwt_secret)
+        end
+
+        it 'returns an error' do
```

Comment:
We don't allow users to access offers in the draft state. I _think_ this is okay, but should we restrict other states too?

Reply:
These are the states on an offer: 

```
  STATES = [
    DRAFT = 'draft',
    SENT = 'sent',
    ACCEPTED = 'accepted',
    REJECTED = 'rejected',
    LAPSED = 'lapsed',
    REVIEW = 'review'
  ].freeze
```

Aside from "draft", all of them seem like states that, if I were a consignor, I might want to reach out about if I'd changed my mind or had questions. I feel like we're good allowing responses in any states except "draft".